tests/proc.test.ts

```diff
@@ -0,0 +1,72 @@
+import { StoredProcedure, StoredProcedureContext, TestingRuntime, Workflow, WorkflowContext, WorkflowHandle } from "../src";
+import { DBOSConfig } from "../src/dbos-executor";
+import { createInternalTestRuntime } from "../src/testing/testing_runtime";
+import { TestKvTable, generateDBOSTestConfig, setUpDBOSTestDb } from "./helpers";
+
+class ProcTest {
+    @StoredProcedure()
+    static async testProc(ctx: StoredProcedureContext, name: string): Promise<string> {
+        const { rows } = await ctx.query<TestKvTable>(`INSERT INTO dbos_test_kv(id, value) VALUES (1, $1) RETURNING id`, [name]);
+        return `hello ${rows[0].id}`;
+    }
+
+    @Workflow()
+    static async testWorkflow(ctx: WorkflowContext, name: string): Promise<string> {
+        return ctx.invoke(ProcTest).testProc(name);
+    }
+}
+
+describe("stored-proc-tests", () => {
+    let username: string;
+    let config: DBOSConfig;
+    let testRuntime: TestingRuntime;
+
+    beforeAll(async () => {
+        config = generateDBOSTestConfig();
+        username = config.poolConfig.user || "postgres";
+        await setUpDBOSTestDb(config);
+    });
+
+    beforeEach(async () => {
+        testRuntime = await createInternalTestRuntime([ProcTest], config);
+        await testRuntime.queryUserDB(create);
+    });
+
+    afterEach(async () => {
+        await testRuntime.queryUserDB(drop);
+        await testRuntime.destroy();
+    });
+
+    test("simple-function", async () => {
+        const workflowHandle: WorkflowHandle<string> = await testRuntime.startWorkflow(ProcTest).testWorkflow(username);
+        const workflowResult: string = await workflowHandle.getResult();
+        expect(workflowResult).toEqual("hello 1");
+      });
+})
+
+const create = `
+CREATE OR REPLACE PROCEDURE "testProc_proc"(
```

Comment:
We need more elaborate tests, especially for the safety code in the liquid templates. For example, we need a test for the correctness of the output buffering code.

Reply:
We need a full end to end test here too IMO